UnitsNet.Tests/CustomCode/DoseAreaProductTests.cs

```diff
@@ -0,0 +1,48 @@
+// <auto-generated>
+//     This code was generated by \generate-code.bat.
+//
+//     Changes to this file will be lost when the code is regenerated.
+//     The build server regenerates the code before each build and a pre-build
+//     step will regenerate the code on each local build.
+//
+//     See https://github.com/angularsen/UnitsNet/wiki/Adding-a-New-Unit for how to add or edit units.
+//
+//     Add CustomCode\Quantities\MyQuantity.extra.cs files to add code to generated quantities.
+//     Add UnitDefinitions\MyQuantity.json and run generate-code.bat to generate new units or quantities.
+//
+// </auto-generated>
+//------------------------------------------------------------------------------
+
+// Licensed under MIT No Attribution, see LICENSE file at the root.
+// Copyright 2013 Andreas Gullberg Larsen (andreas.larsen84@gmail.com). Maintained at https://github.com/angularsen/UnitsNet.
+
+using System;
+
+namespace UnitsNet.Tests.CustomCode
+{
+    public class DoseAreaProductTests : DoseAreaProductTestsBase
+    {
+        protected override bool SupportsSIUnitSystem => true;
+        protected override double CentigrayPerSquareCentimetersInOneGrayPerSquareMeter => 1E+6;
+        protected override double CentigrayPerSquareDecimetersInOneGrayPerSquareMeter => 1E+4;
+        protected override double CentigrayPerSquareMetersInOneGrayPerSquareMeter => 100;
+        protected override double CentigrayPerSquareMillimetersInOneGrayPerSquareMeter => 1E+8;
+        protected override double DecigrayPerSquareCentimetersInOneGrayPerSquareMeter => 1E+5;
+        protected override double DecigrayPerSquareDecimetersInOneGrayPerSquareMeter => 1E+3;
+        protected override double DecigrayPerSquareMetersInOneGrayPerSquareMeter => 10;
+        protected override double DecigrayPerSquareMillimetersInOneGrayPerSquareMeter => 1E+7;
+        protected override double GrayPerSquareCentimetersInOneGrayPerSquareMeter => 1E+4;
+        protected override double GrayPerSquareDecimetersInOneGrayPerSquareMeter => 1E+2;
+        protected override double GrayPerSquareMetersInOneGrayPerSquareMeter => 1;
+        protected override double GrayPerSquareMillimetersInOneGrayPerSquareMeter => 1E+6;
+        protected override double MicrograyPerSquareCentimetersInOneGrayPerSquareMeter => 1E+10;
+        protected override double MicrograyPerSquareDecimetersInOneGrayPerSquareMeter => 1E+8;
+        protected override double MicrograyPerSquareMetersInOneGrayPerSquareMeter => 1E+6;
+        protected override double MicrograyPerSquareMillimetersInOneGrayPerSquareMeter => 1E+12;
+        protected override double MilligrayPerSquareCentimetersInOneGrayPerSquareMeter => 1E+7;
+        protected override double MilligrayPerSquareDecimetersInOneGrayPerSquareMeter => 1E+5;
+        protected override double MilligrayPerSquareMetersInOneGrayPerSquareMeter => 1E+3;
+        protected override double MilligrayPerSquareMillimetersInOneGrayPerSquareMeter => 1E+9;
```

Comment:
Verified in https://www.dosewizard.com/2011/06/dap-converter.html

Reply:
Or all units can be checked [here](http://runphym.free.fr/NewFiles/tbfr-App-Runphym/dap-cur/dapcnv.html).
Actually, due to the area, the formula is easy to derive for any SI prefix and unit, even for inches. But as a rule, square centimeters are widely used for area. I decided to add a little more options